molgenis-questionnaires/src/main/frontend/src/store/actions.js

```diff
@@ -0,0 +1,77 @@
+import api from '@molgenis/molgenis-api-client'
+
+import { EntityToFormMapper } from '@molgenis/molgenis-ui-form'
+
+const actions = {
+  'GET_QUESTIONNAIRE_LIST' ({commit}) {
+    return api.get('/menu/plugins/questionnaires/list').then(response => {
+      commit('SET_QUESTIONNAIRE_LIST', response)
+    })
+  },
+
+  'START_QUESTIONNAIRE' ({commit, state}, questionnaireId) {
+    if (state.questionnaireId !== questionnaireId) {
+      commit('CLEAR_STATE')
+    }
+
+    api.get('/menu/plugins/questionnaires/start/' + questionnaireId)
+  },
+
+  'GET_QUESTIONNAIRE' ({state, commit}, questionnaireId) {
+    return api.get('/api/v2/' + questionnaireId + '?includeCategories=true').then(response => {
+      commit('SET_QUESTIONNAIRE_ID', questionnaireId)
+      commit('SET_QUESTIONNAIRE_LABEL', response.meta.label)
+      commit('SET_QUESTIONNAIRE_DESCRIPTION', response.meta.description)
+
+      const data = response.items.length > 0 ? response.items[0] : {}
+      const form = EntityToFormMapper.generateForm(response.meta, data, state.mapperOptions)
+      const chapters = form.formFields.filter(field => field.type === 'field-group')
+
+      commit('SET_CHAPTER_FIELDS', chapters)
+      commit('SET_QUESTIONNAIRE_ROW_ID', data[response.meta.idAttribute])
+      commit('SET_FORM_DATA', form.formData)
+    })
+  },
+
+  /* eslint-disable no-empty-pattern */
+  'GET_QUESTIONNAIRE_OVERVIEW' ({}, questionnaireId) {
+    return api.get('/api/v2/' + questionnaireId)
```

Comment:
why not use the store ?

Reply:
I don't understand the comment. Overview screen is a separate readonly view of one row in the questionnaire table. If linked via URL, the overview component will need to fetch data. It could even work outside of the store.